### ISSUE TITLE
Document how to use the combinator as a client..

### ISSUE THREAD

**Issue by alexanderkjeldaas** [haskell-servant/servant-multipart#3]
.. or change the description to indicate that this is only server-side.

**Comment by alpmestan** [comment]
It indeed only supports the server interpretation at the moment. If someone feels like implementing the client interpretation, I'm of course willing to help, but don't need it myself. https://hackage.haskell.org/package/http-client-0.5.6.1/docs/Network-HTTP-Client-MultipartFormData.html could be used to that effect but I don't know whether that would require any change to servant-client or not.

**Comment by jonathanjouty** [comment]
#26 seems to have (partially? fully?) resolved this, but has not cut into a new version yet.

**Comment by alpmestan** [comment]
Yes #26 solves this, and we should make a new release! I'll try to make that happen in the upcoming days.

**Comment by jonathanjouty** [comment]
Great, thanks @alpmestan, and Hello! :)

Before a new release is made how do you feel about me making [`genBoundary`](https://github.com/haskell-servant/servant-multipart/blob/master/src/Servant/Multipart.hs#L325) not depend on `IO`?
Either by taking a `StdGen` or using [`MonadRandom`](https://hackage.haskell.org/package/MonadRandom-0.5.1.1/docs/Control-Monad-Random-Class.html#t:MonadRandom), or anything more abstract than plain `IO` really!

I am happy to take suggestions on this...

**Comment by alpmestan** [comment]
@jonathanjouty I don't think I'd have any problem with that, but I'm curious, what's the motivation?

As for the concrete way to make it happen, well I think if you just remove the call to `getStdRandom` (and possibly adapt the rest of the code to make things typecheck), then what you get back is something that's parametrised over the concrete `RandomGen` implementation, isn't it? That should be fairly straightforward, and since that `genBoundary` is supposed to be a helper that users need to use explicitly, I don't think it'll be a problem to make it work in more contexts.

Perhaps you can provide a name for the variant that picks `StdGen` and returns a boundary in `IO`? I'll just let you think about the best "presentation" for this change, since you're the one with a concrete use case in his hands.

**Comment by jonathanjouty** [comment]
> since you're the one with a concrete use case in his hands.

Actually, I imagined it! Sorry!
The implementation is perfectly fine as it is :)
